grafana-plugin/src/containers/IntegrationForm/IntegrationForm.tsx

```diff
@@ -32,481 +32,613 @@ import { IntegrationHelper } from 'pages/integration/Integration.helper';
-  const [selectedOption, setSelectedOption] = useState<ApiSchemas['AlertReceiveChannelIntegrationOptions']>(undefined);
-  const [showIntegrarionsListDrawer, setShowIntegrarionsListDrawer] = useState(id === 'new');
-  const [allContactPoints, setAllContactPoints] = useState([]);
-  const [errors, setErrors] = useState<Record<string, any>>();
-
-  const form = useMemo(() => getForm(grafanaTeamStore), [grafanaTeamStore]);
-
-  useEffect(() => {
-    (async function () {
-      setAllContactPoints(await AlertReceiveChannelHelper.getGrafanaAlertingContactPoints());
-    })();
-  }, []);
-
-  const data =
-    id === 'new'
-      ? { integration: selectedOption?.value, team: user?.current_team, labels: [] }
+const RADIO_OPTIONS = [
+  {
+    label: 'Connect existing Contact point',
+    value: 'existing',
+  },
+  {
+    label: 'Create a new one',
+    value: 'new',
+  },
+];
+
+export const IntegrationForm = observer(
+  ({
+    id,
+    isTableView,
+    navigateToAlertGroupLabels,
+    selectedIntegration,
+    onSubmit,
+    onHide,
+    onBackClick,
+  }: IntegrationFormProps) => {
+    const store = useStore();
+    const history = useHistory();
+    const styles = useStyles2(getIntegrationFormStyles);
+    const isNew = id === 'new';
+    const { userStore, grafanaTeamStore, alertReceiveChannelStore } = store;
+
+    const data = isNew
+      ? { integration: selectedIntegration?.value, team: userStore.currentUser?.current_team, labels: [] }
       : prepareForEdit(alertReceiveChannelStore.items[id]);
 
-  const { alertReceiveChannelOptions } = alertReceiveChannelStore;
-
-  const options = alertReceiveChannelOptions
-    ? alertReceiveChannelOptions.filter((option: ApiSchemas['AlertReceiveChannelIntegrationOptions']) => {
-        if (option.value === 'grafana_alerting' && !window.grafanaBootData.settings.unifiedAlertingEnabled) {
-          return false;
-        }
-
-        // don't allow creating direct paging integrations
-        if (option.value === 'direct_paging') {
-          return false;
-        }
-
-        return (
-          option.display_name.toLowerCase().includes(filterValue.toLowerCase()) &&
-          !option.value.toLowerCase().startsWith('legacy_')
-        );
-      })
-    : [];
-
-  const extraGFormProps: { customFieldSectionRenderer?: React.FC<CustomFieldSectionRendererProps> } = {};
+    const { integration } = data;
+
+    const formMethods = useForm<FormFields>({
+      defaultValues: isNew
+        ? {
+            // these are the default values for creating an integration
+            [FormFieldKeys.Integration]: integration,
+            [FormFieldKeys.DefaultWebhooks]: true,
+          }
+        : {
+            // existing values from existing integration (edit-mode)
+            ...data,
+          },
+      mode: 'all',
+    });
+
+    const {
+      control,
+      handleSubmit,
+      formState: { errors },
+    } = formMethods;
```

Comment:
I guess you can destructure right away in line 115

Reply:
There's been changes, which is line 115? :laughing: you meant right after it? done

Reply:
I mean 
```
const {
      control,
      handleSubmit,
      formState: { errors },
    } = useForm<FormFields>({
    ......
```

Reply:
I pass `formMethods` to the `FormProvider`. If I destructure and discard the usage of `formMethods` I'll then need to pass each destructured property...

Reply:
missed that, all good